pkg/tracing/opentelemetry/opentelemetry.go

```diff
@@ -0,0 +1,135 @@
+package opentelemetry
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/url"
+	"time"
+
+	"github.com/webhookx-io/webhookx/config"
+	"go.opentelemetry.io/contrib/propagators/autoprop"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.26.0"
+	"go.opentelemetry.io/otel/trace"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/encoding/gzip"
+)
+
+const TracerName = "github.com/webhookx-io/webhookx"
+
+type OpentelemetryConfig config.OpenTelemetryConfig
+
+func (o *OpentelemetryConfig) Setup(serviceName string, samplingRate float64, globalAttributes map[string]string) (trace.Tracer, io.Closer, error) {
```

Comment:
I prefer to make OpentelemetryConfig as a function arg instead of a receiver.

Reply:
interface should be better to implement different tracing setup.